File: baekjoon/12865.py
import sys
input = sys.stdin.readline
n, k = map(int, input().split())
lst = [[0, 0]]
for _ in range(1, n+1):
    lst.append(list(map(int, input().split())))

dp = [[0]*(k+1) for _ in range(n+1)]
for i in range(1, n+1):     # 가방 개수 만큼 반복
    for j in range(1, k+1): # 최대 무게까지 반복
        w = lst[i][0]
        v = lst[i][1]
        if j < w: dp[i][j] = dp[i-1][j]     # 가방에 넣을 수 없음
        else:
            dp[i][j] = max(dp[i-1][j], v + dp[i-1][j-w])    # 가방에 넣을 수 있음
print(dp[n][k])

File: baekjoon/12865.js
// https://www.acmicpc.net/problem/12865
// 평범한 배낭
// dp

const fs = require("fs");
const filePath = process.platform === "linux" ? "/dev/stdin" : "./input.txt";
const input = fs.readFileSync(filePath).toString().trim().split("\n");

const [n, k] = input.shift().split(" ").map(Number);
const items = input.map((item) => item.split(" ").map(Number));

function solution() {
  let dp = Array.from(Array(n + 1), () => Array(k + 1).fill(0));
  for (let i = 1; i < n + 1; i++) {
    let w = items[i - 1][0];
    let v = items[i - 1][1];
    for (let j = 1; j < k + 1; j++) {
      if (w <= j) {
        dp[i][j] = Math.max(dp[i - 1][j], v + dp[i - 1][j - w]);
      } else {
        dp[i][j] = dp[i - 1][j];
      }
    }
  }
  console.log(dp[n][k]);
}

solution();
